File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messageitem/Quote.java
package forpleuvoir.dhwu1a.core.message.messageitem;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType;

import java.util.List;

import static forpleuvoir.dhwu1a.core.websocket.base.ApiKey.*;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messageitem
 * <p>#class_name Quote
 * <p>#create_time 2021/6/30 00:10
 */
public class Quote extends MessageItem {
    /**
     * 被引用回复的原消息的messageId
     */
    @SerializedName(ID)
    public final long id;
    /**
     * 被引用回复的原消息所接收的群号，当为好友消息时为0
     */
    @SerializedName(GROUP_ID)
    public final long groupId;
    /**
     * 被引用回复的原消息的发送者的QQ号
     */
    @SerializedName(SENDER_ID)
    public final long senderId;
    /**
     * 被引用回复的原消息的接收者者的QQ号（或群号）
     */
    @SerializedName(TARGET_ID)
    public final long targetId;
    /**
     * 被引用回复的原消息的消息链对象
     */
    @SerializedName(ORIGIN)
    public final List<Plain> origin;

    public Quote(long id, long groupId, long senderId, long targetId,
                 List<Plain> origin
    ) {
        super(MessageItemType.Quote);
        this.id = id;
        this.groupId = groupId;
        this.senderId = senderId;
        this.targetId = targetId;
        this.origin = origin;
    }

    @Override
    public String toPlainText() {
        return String.format("回复[%s]:", getOriginPlainText());
    }

    private String getOriginPlainText() {
        StringBuilder stringBuilder = new StringBuilder();
        for (Plain plain : origin) {
            stringBuilder.append(plain.toPlainText());
        }
        return stringBuilder.toString();
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messageitem/Quote.kt
package forpleuvoir.dhwu1a.core.message.messageitem

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.*
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messageitem
 *
 * #class_name Quote
 *
 * #create_time 2021/6/30 00:10
 */
class Quote(
    /**
     * 被引用回复的原消息的messageId
     */
    @field:SerializedName(ID) val id: Long,
    /**
     * 被引用回复的原消息所接收的群号，当为好友消息时为0
     */
    @field:SerializedName(GROUP_ID) val groupId: Long,
    /**
     * 被引用回复的原消息的发送者的QQ号
     */
    @field:SerializedName(SENDER_ID) val senderId: Long,
    /**
     * 被引用回复的原消息的接收者者的QQ号（或群号）
     */
    @field:SerializedName(TARGET_ID) val targetId: Long,
    /**
     * 被引用回复的原消息的消息链对象
     */
    @field:SerializedName(ORIGIN) val origin: List<Plain>
) : MessageItem(MessageItemType.Quote) {
    override fun toPlainText(): String {
        return String.format("回复[%s]:", originPlainText)
    }

    private val originPlainText: String
        get() {
            val stringBuilder = StringBuilder()
            for (plain in origin) {
                stringBuilder.append(plain.toPlainText())
            }
            return stringBuilder.toString()
        }
}